packages/mobile/src/identity/feelessVerification.ts

```diff
@@ -0,0 +1,1061 @@
+            attestationsToRequest,
+          })
+          attestations = yield call(
+            requestAndRetrieveAttestations,
+            attestationsWrapper,
+            phoneHash,
+            unverifiedMtwAddress,
+            attestations,
+            attestations.length + attestationsToRequest,
+            true,
+            komenciKit
+          )
+          ValoraAnalytics.track(VerificationEvents.verification_request_all_attestations_complete, {
+            issuers,
+          })
+
+          // start listening for the new list of attestations
+          receiveMessageTask?.cancel()
+          issuers = attestations.map((a) => a.issuer)
+          receiveMessageTask = yield takeEvery(
+            Actions.RECEIVE_ATTESTATION_MESSAGE,
+            attestationCodeReceiver(
+              attestationsWrapper,
+              phoneHash,
+              unverifiedMtwAddress,
+              issuers,
+              true
+            )
+          )
+
+          // Request codes for the new list of attestations. We ignore unsuccessfull reveals here,
+          // cause we do not want to go into a loop of re-requesting more and more attestations
+          yield call(
+            revealAttestations,
+            attestationsWrapper,
+            unverifiedMtwAddress,
+            phoneHashDetails,
+            attestations,
+            true
+          )
+        }
+        ValoraAnalytics.track(VerificationEvents.verification_reveal_all_attestations_complete)
+      }
+
+      yield put(feelessSetVerificationStatus(VerificationStatus.CompletingAttestations))
+
+      yield race([
+        call(
+          completeAttestations,
+          attestationsWrapper,
+          unverifiedMtwAddress,
+          phoneHashDetails,
+          attestations,
+          true,
+          komenciKit
+        ),
+        // This is needed, because we can have more actionableAttestations than NUM_ATTESTATIONS_REQUIRED
+        call(feelessRequiredAttestationsCompleted),
+      ])
+
+      receiveMessageTask?.cancel()
+      if (Platform.OS === 'android') {
+        autoRetrievalTask?.cancel()
+      }
+    }
+
+    // Intentionally calling this a second time. No-op if they already registered so no harm done
+    yield call(feelessDekAndWalletRegistration, komenciKit, walletAddress)
+    yield call(endFeelessVerification)
+    return true
+  } catch (error) {
+    Logger.error(TAG, 'Error occured during feeless verification flow', error)
+    yield all([
+      call(storeTimestampIfKomenciError, error, true),
+      put(feelessSetVerificationStatus(VerificationStatus.Failed)),
+      put(showErrorOrFallback(error, ErrorMessages.VERIFICATION_FAILURE)),
+    ])
+    return error.message
+  } finally {
+    receiveMessageTask?.cancel()
+    if (Platform.OS === 'android') {
+      autoRetrievalTask?.cancel()
+    }
+  }
+}
+
+function* fetchKomenciReadiness(komenciKit: KomenciKit) {
```

Comment:
IIUC this function is supposed to update the state about whether Komenci is available. Should it update the state to false if the service check fails?